draft-ietf-acme-acme.md

```diff
@@ -484,29 +484,49 @@ server MAY return status code 405 (Method Not Allowed).
 | malformed             | The request message was malformed                                  |
 | rateLimited           | The request exceeds a rate limit                                   |
 | rejectedIdentifier    | The server will not issue for the identifier                       |
 | serverInternal        | The server experienced an internal error                           |
-| tls                   | The server received a TLS error during validation                  |
 | unauthorized          | The client lacks sufficient authorization                          |
-| unknownHost           | The server could not resolve a domain name                         |
-| unsupportedIdentifier | Identifier is not supported but may be in the future               |
-| userActionRequired    | The user visit the "instance" URL and take actions specified there |
+| unsupportedIdentifier | Identifier is not supported, but may be in future                  |
+| userActionRequired    | Visit the "instance" URL and take actions specified there          |
 | badRevocationReason   | The revocation reason provided is not allowed by the server        |
 
-This list is not exhaustive. The server MAY return errors whose "type" field is
+Error types that may be commonly included in the "error" field of challenge
+resources:
+
+| Type                  | Description                                                        |
+|:----------------------|:-------------------------------------------------------------------|
+| caa                   | CAA records forbid the CA from issuing                             |
+| connection            | The server could not connect to validation target                  |
+| dns                   | There was a problem with a DNS query                               |
+| rateLimited           | The request exceeds a rate limit                                   |
+| serverInternal        | The server experienced an internal error                           |
+| incorrectResponse     | Response received didn't match the challenge's requirements        |
+| tls                   | The server received a TLS error during validation                  |
+
+Error types that may be commonly included in the "error" field of order
+resources:
+
+| Type                  | Description                                                        |
+|:----------------------|:-------------------------------------------------------------------|
+| caa                   | CAA records forbid the CA from issuing                             |
+| dns                   | There was a problem with a DNS query                               |
+| unsupportedIdentifier | Identifier is not supported but may be in the future               |
+| serverInternal        | The server experienced an internal error                           |
+
+These lists are not exhaustive. The server MAY return errors whose "type" field is
```

Comment:
This change is problematic.  You've taken a normative list of all error codes to be registered for this protocol (and which IANA is going to use to initialize a registry) and turned it into three tables of examples. Instead, this section should list all error codes, and the specific lists of examples should move to the corresponding sections.